yjit/src/codegen.rs

```diff
@@ -4996,46 +5000,44 @@ fn gen_invokebuiltin(jit: &mut JITState, ctx: &mut Context, cb: &mut CodeBlock,
 // stack uses the argument locals (and self) from the current method.
 fn gen_opt_invokebuiltin_delegate(jit: &mut JITState, ctx: &mut Context, cb: &mut CodeBlock, ocb: &mut OutlinedCb) -> CodegenStatus
 {
-    const struct rb_builtin_function *bf = (struct rb_builtin_function *)jit_get_arg(jit, 0);
-    int32_t start_index = (int32_t)jit_get_arg(jit, 1);
+    let bf: *const rb_builtin_function = jit_get_arg(jit, 0).as_ptr();
+    let bf_argc = unsafe { (*bf).argc };
+    let start_index = jit_get_arg(jit, 1).as_i32();
 
     // ec, self, and arguments
-    if (bf->argc + 2 > NUM_C_ARG_REGS) {
+    if bf_argc + 2 > (C_ARG_REGS.len().try_into().unwrap()) {
```

Comment:
I feel like it's very verbose to use `try_into().unwrap()` everywhere. In most cases we know the value can't overflow so it would be fine to use `as`. In this specific case the array length is a constant.